kimchi/src/circuits/polynomials/foreign_field_mul/circuitgates.rs

```diff
@@ -282,30 +200,34 @@ where
             )
         );
 
-        // Compute intermediate sums
-        let [sum01, sum2] = compute_intermediate_sums(&quotient, &neg_foreign_field_modulus);
-
         // Compute native modulus values
-        let [left_input_n, right_input_n, quotient_n, remainder_n, foreign_field_modulus_n] =
+        let [left_input_n, right_input_n, quotient_n, remainder_n, neg_foreign_field_modulus_n] =
             compute_native_modulus_values(
                 &left_input,
                 &right_input,
                 &quotient,
                 &remainder,
-                &foreign_field_modulus,
+                &neg_foreign_field_modulus,
             );
 
+        // bound = x2 + 2^88 - f2 - 1
+        let bound = quotient[2].clone() + T::two_to_limb() - hi_foreign_field_modulus - T::one();
+
         // Define the constraints
         //   For more the details on each constraint please see the
         //   Foreign Field Multiplication RFC where each of the constraints
         //   numbered below are described in full detail.
 
+        // External checks
+        // multi-range-check: q'2, p10, p110
+        // That is, check bound, product1_lo, product1_hi_0 each in [0, 2^L)
+        // Must be done externally with a multi-range-check gadget
+
         // C1: Constrain intermediate product fragment product1_hi_1 \in [0, 2^2)
         constraints.push(product1_hi_1.crumb());
```

Comment:
The reason that they are indeed with a different number, is because tests of the gate expect a constraint number in order to check if that one works or not. Thus, I found it useful to just use the code numbering to match the framework indices. But I can write in parenthesis next to each, the correspondence with the RFC if that makes you feel more convinced.